benchmark_runner/common/remote_ssh/remote_ssh.py

```diff
@@ -0,0 +1,195 @@
+        self.__p_client.set_missing_host_key_policy(
+            paramiko.AutoAddPolicy())  # This script doesn't work for me unless this line is added!
+
+        self.__p_sftp = None
+        # remote host connection data
+        self.__hostname = connection_data.host_name
+        self.__port = connection_data.port
+        self.__username = connection_data.user_name
+        self.__key_file = connection_data.ssh_key
+        self.__password = connection_data.password
+        self.__cmd_timeout = connection_data.timeout
+
+    def connect(self):
+        """ This method connect to remote machine """
+        try:
+            # For connection with ssh key
+            if self.__key_file:
+                key = paramiko.RSAKey.from_private_key_file(self.__key_file)
+                self.__p_client.connect(hostname=self.__hostname, port=self.__port, username=self.__username,
+                                        pkey=key, timeout=self.__cmd_timeout)
+            # For password
+            elif self.__password:
+                self.__p_client.connect(hostname=self.__hostname, port=self.__port, username=self.__username,
+                                        password=self.__password, timeout=self.__cmd_timeout)
+            else:
+                raise SshConnectionError
+
+            self.__p_sftp = self.__p_client.open_sftp()
+
+        except socket.timeout as err:
+            raise SshConnectionTimedOut(err)
+
+        except Exception as err:
+            raise SshConnectionFailure(err)
+
+    def disconnect(self):
+        """ This method disconnect from remote machine """
+        self.__p_client.close()
+        self.__p_sftp.close()
+
+    def run_command(self, command: str):
+        """
+        This method run command
+        :param command:
+        :return: command response or Exception
+        """
+        try:
+            pre_command = '. ~/.bash_profile;'
+            stdin, stdout, stderr = self.__p_client.exec_command(f'{pre_command}\n {command}')
+            opt = stdout.readlines()
+            message = "".join(opt)
+            return message
+        except Exception as err:
+            raise RunCommandError(err)
+
+    def mkdir(self, remote_path):
+        """
+        This method mkdir on remote path
+        :param remote_path:
+        :return:
+        """
+        try:
+            self.__p_sftp.chdir(remote_path)  # Test if remote path exists
+        except IOError:
+            self.__p_sftp.mkdir(remote_path)  # Create remote path
+            self.__p_sftp.chdir(remote_path)
+
+    def put_remote_dir(self, local, remote):
+        """
+        This method uploads the contents of the local directory to the remote path. The
+        remote directory needs to exists. All subdirectories in py_perf-code are
+        created under target.
+        :param local: local path
+        :param remote: remote path
+        :return:
+        """
+        for file_name in os.listdir(local):
```

Comment:
What if some of the files aren't plain files (directories, sockets, symlinks, etc)?